mods/sib-plusplus-tweaker.wh.cpp

```diff
@@ -0,0 +1,453 @@
+{
+  _GUID *folderID;
+  const wchar_t *strParseName;
+  const wchar_t *strSettingKey;
+  int iShowDefault;
+  const wchar_t *strPolicyOff;
+  const wchar_t *strPolicyOn;
+  const wchar_t *desktopIconCLSID;
+  SHELLMENUTYPE iMenuType;
+  unsigned int resNameId;
+  unsigned int resTipId;
+  wchar_t fontImage;
+};
+
+HMODULE g_hStartIsBackModule = nullptr;
+ULONGLONG g_StartIsBackSize = 0;
+
+FOLDERDEFINITION* SpecialFoldersList;
+#define REL(addr, offset) ((addr + offset + 4) + *(int32_t*)(addr + offset))
+#define OFF(addr, offset) addr = addr + offset;
+
+static uintptr_t FindPattern(const char* signature, uintptr_t base_address, int Relative = 0, int Offset = 0)
+{
+    static auto patternToByte = [](const char* pattern)
+    {
+        auto bytes = std::vector<int>{};
+        const auto start = const_cast<char*>(pattern);
+        const auto end = const_cast<char*>(pattern) + strlen(pattern);
+        for (auto current = start; current < end; ++current)
+        {
+            if (*current == '?')
+            {
+                ++current;
+                if (*current == '?')
+                    ++current;
+                bytes.push_back(-1);
+            }
+            else { bytes.push_back(strtoul(current, &current, 16)); }
+        }
+        return bytes;
+    };
+    const auto dosHeader = (PIMAGE_DOS_HEADER)base_address;
+    const auto ntHeaders = (PIMAGE_NT_HEADERS)((std::uint8_t*)base_address + dosHeader->e_lfanew);
+    const auto sizeOfImage = ntHeaders->OptionalHeader.SizeOfImage;
+    auto patternBytes = patternToByte(signature);
+    const auto scanBytes = reinterpret_cast<std::uint8_t*>(base_address);
+    const auto s = patternBytes.size();
+    const auto d = patternBytes.data();
+    for (auto i = 0ul; i < sizeOfImage - s; ++i)
+    {
+        bool found = true;
+        for (auto j = 0ul; j < s; ++j)
+        {
+            if (scanBytes[i + j] != d[j] && d[j] != -1)
+            {
+                found = false;
+                break;
+            }
+        }
+        if (found)
+        {
+            uintptr_t address = reinterpret_cast<uintptr_t>(&scanBytes[i]);
+            if (Relative)
+            {
+                REL(address, Relative);
+            }
+            if (Offset)
+            {
+                OFF(address, Offset);
+            }
+            return address;
+        }
+    }
+    return NULL;
+}
+
+void DoModulePatch()
+{
+    Wh_Log(L"Attempting to:");
+
+    DWORD old;
+    if(mod_settings.DHTracking == TRUE)
+    {
+        Wh_Log(L"- Disable Hottracking...");
+        uint8_t* p_trackingBG = (uint8_t*)(FindPattern("48 8B C4 48 89 50 10 48 89 48 08",(uintptr_t)g_hStartIsBackModule));
+        uint8_t* p_trackingBorder = (uint8_t*)(FindPattern("4C 8B DC 49 89 6B 18 49 89 73 20",(uintptr_t)g_hStartIsBackModule));
+
+        VirtualProtect(p_trackingBG, 1, PAGE_EXECUTE_READWRITE, &old);
+        p_trackingBG[0] = 0xC3;
+        VirtualProtect(p_trackingBG, 1, old, 0);
```

Comment:
Does this work?
[From the documentation](https://learn.microsoft.com/en-us/windows/win32/api/memoryapi/nf-memoryapi-virtualprotect) for the last argument:
> If this parameter is NULL or does not point to a valid variable, the function fails.

Reply:
This works fine. I made sure to not run it if it isn't properly setup. Might have to add a fallback incase the pattern isn't found by whatever reason.

Reply:
It might work in practice, but if the second `VirtualProtect` call fails, the memory will always stay with `PAGE_EXECUTE_READWRITE`.

According to the documentation it doesn't work. I'd replace `VirtualProtect(p_trackingBG, 1, old, 0);` with `VirtualProtect(p_trackingBG, 1, old, &old);` here and in the other places.

Reply:
Noted. Will do.